service/internal/integrationtest/s3shared/integ_test_setup.go

```diff
@@ -0,0 +1,134 @@
+// +build integration
+
+package s3shared
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/internal/integrationtest"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3control"
+)
+
+// BucketPrefix is the root prefix of integration test buckets.
+const BucketPrefix = "aws-sdk-go-v2-integration"
+
+// GenerateBucketName returns a unique bucket name.
+func GenerateBucketName() string {
+	return fmt.Sprintf("%s-%s",
+		BucketPrefix, integrationtest.UniqueID())
+}
+
+// SetupBucket returns a test bucket created for the integration tests.
+func SetupBucket(ctx context.Context, svc *s3.Client, bucketName string) (err error) {
+	fmt.Println("Setup: Creating test bucket,", bucketName)
+	_, err = svc.CreateBucket(ctx, &s3.CreateBucketInput{
+		Bucket: &bucketName,
+	})
+	if err != nil {
+		fmt.Errorf("failed to create bucket %s, %v", bucketName, err)
+	}
+
+	// TODO: change this to use waiter to wait until BucketExists instead of sleep timer
+	// 	svc.WaitUntilBucketExists(HeadBucketInput)
+	time.Sleep(1 * time.Minute)
```

Comment:
Instead of sleeping this should just do a loop for `HeadBucket` until the bucket exists.  Similar for other waiters that were removed. This will prevent extra waited time and probably more stable tests.